babelsberg/constraintinterpreter.js

```diff
@@ -258,63 +257,31 @@ Object.subclass('Babelsberg', {
         func.onError = opts.onError;
 
         solvers = this.filterSolvers(solvers, opts);
-
-        solvers.each(function(solver) {
+        var constraints = this.createEquivalentConstraints(solvers, opts, func, errors);
+        var constraint = this.chooseConstraint(constraints, opts, errors);
+        if (!opts.postponeEnabling && constraint) {
             try {
-                var constraint = solver.always(Object.clone(opts), func);
-                constraint.logTimings = opts.logTimings;
-                constraints.push(constraint);
-            } catch (e) {
-                errors.push(e);
-                return false;
-            }
-        });
-
-        if (constraints.length > 1) {
-            for (var i = 0; i < constraints.length; i++) {
+                constraint.isAnyVariableCurrentlySuggested = true; // do not increase
+                                                                   // updateCounter
                 try {
-                    Constraint.current = constraints[i];
-                    constraints[i].enable(true);
-                    constraints[i].disable();
-                } catch (e) {
-                    errors.push(e);
-                    constraints[i].disable();
-                    constraints[i] = null;
+                    constraint.enable();
                 } finally {
-                    Constraint.current = null;
-                }
-            }
-
-            var min = Number.MAX_VALUE;
-            var minIndex = -1;
-            for (var i = 0; i < constraints.length; i++) {
-                if (constraints[i] && constraints[i].oComparisonMetrics.time < min) {
-                    min = constraints[i].oComparisonMetrics.time;
-                    minIndex = i;
+                    constraint.isAnyVariableCurrentlySuggested = false;
                 }
-            }
-            if (minIndex > -1) {
-                constraint = constraints[minIndex];
-                if (opts.logTimings) {
-                    console.log('Selected fastest solver:' +
-                        constraint.solver.solverName);
-                }
-            }
-        } else if (constraints.length == 1) {
-            constraint = constraints[0];
-        }
-
-        if (!opts.postponeEnabling && constraint) {
-            try {
-                constraint.enable();
             } catch (e) {
                 errors.push(e);
                 constraint.disable();
+                constraint.isAbandoned = true;
                 constraint = null;
             }
         }
-
-        if (!constraint) {
+        if (constraint) {
+            constraints.each(function(each) {
+                if (each !== constraint && each !== null)
+                    each.isAbandoned = true;
+            });
+            constraint.compactExternalVariablesOfConstrainedVariables();
```

Comment:
just nitpicking here, but from a flow and readability perspective I'd like `constraint.abandon()` rather than setting individual booleans and then calling some compactor